terratorch/models/prithvi_model_factory.py

```diff
@@ -95,7 +98,13 @@ def build_model(
                 msg = f"Task {task} not supported. Please choose one of {SUPPORTED_TASKS}"
                 raise NotImplementedError(msg)
 
+            # These params are used in case we need a SMP decoder
+            # but should not be used for timm encoder
             backbone_kwargs, kwargs = _extract_prefix_keys(kwargs, "backbone_")
+            smp_kwargs, kwargs = _extract_prefix_keys(kwargs, "smp_")
+            aux_kwargs, kwargs = _extract_prefix_keys(kwargs, "aux_")
```

Comment:
Can we make use of the args passed through `decoder_` instead of needing a new prefix for smp and aux?